### PR TITLE
Converts the Scan-operation from PushPullStage to GraphStage

### PR DESCRIPTION
**Pull request by viktorklang**
Sanity checks are appreciated @akka/akka-committers

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala [hunk 1]
```diff
@@ -288,49 +284,6 @@ class InterpreterSupervisionSpec extends AkkaSpec with GraphInterpreterSpecKit {
       }
     }
 
-    "resume when Scan throws" in new OneBoundedSetup[Int](Seq(
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala]: Restart is covered by existing test "fail properly" (great name btw), but  not resume?

**Comment by johanandren** [comment]
LGTM, except possibly missing test coverage for resume

**Comment by viktorklang** [comment]
@johanandren Added test for "resume"

**Comment by patriknw** [comment]
thank you, @viktorklang

**Pull request merged**